app/models/submission.rb

```diff
@@ -482,6 +498,268 @@ def turnitinable_by_lti?
     turnitin_data.select{|_, v| v.is_a?(Hash) && v.key?(:outcome_response)}.any?
   end
 
+  # VeriCite
+
+  # this function will check if the score needs to be updated and update/save the new score if so,
+  # otherwise, it just returns the vericite_data_hash
+  def vericite_data()
+    self.vericite_data_hash ||= {}
+    self.vericite_data_hash[:provider] = :vericite
+    # check to see if the score is stale, if so, fetch it again
+    updateScores = false
+    # since there could be multiple versions, let's not waste calls for old versions and use the old score
+    if !self.readonly? && self.versions.current.number == self.version_number
+      self.vericite_data_hash.keys.each do |asset_string|
+        data = self.vericite_data_hash[asset_string]
+        next unless data && data.is_a?(Hash) && data[:object_id]
+        updateScores = updateScores || vericite_recheck_score(data)
+      end
+      # we have found at least one score that is stale, call VeriCite and save the results
+      if updateScores
+        check_vericite_status
+      end
```

Comment:
This could potentially turn into one network call per submission on a large load like speed_grader_json.  This can't happen inline like this.  check_vericite_status should only ever happen when requested from the front-end or from a delayed job after the submissions.